content/java/concepts/array-list/terms/iterator/iterator.md

```diff
@@ -0,0 +1,61 @@
+---
+Title: '.iterator()'
+Description: 'Returns an iterator over the elements in an ArrayList'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Array List'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.iterator()`** method is used to iterate over the elements of a collection (such as an [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list)) one by one. 
+
+## Syntax
+
+```pseudo
+arrayListInstance.iterator();
+```
+
+It returns an [`Iterator`](https://www.codecademy.com/resources/docs/java/iterator) object, which must be of the same [data type](https://www.codecademy.com/resources/docs/java/data-types) as `arrayListInstance`.
+
+## Example
+
+The following example creates an `ArrayList`, inserts some elements using the [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) method and then iterates over it using a loop.
```

Comment:
Modifications:

- Let's use a colon (:) instead of a period (.) as the sentence is introducing an example
- Let's link 'loop' to its corresponding doc entry

```suggestion
The following example creates an `ArrayList`, inserts some elements using the [`.add()`](https://www.codecademy.com/resources/docs/java/array-list/add) method and then iterates over it using a [loop](https://www.codecademy.com/resources/docs/java/loops):
```